docs/source/data/errors.mdx

```diff
@@ -0,0 +1,474 @@
+
+### Including custom error details
+
+Whenever you throw an `ApolloError`, you can add arbitrary fields to the error's `extensions` object to provide additional context to the client. You specify these fields in an object you provide to the error's constructor.
+
+This example builds on the one above by adding the name of the GraphQL argument that was invalid:
+
+<ExpansionPanel title="Click to expand">
+
+```js{22-24}
+const {
+  ApolloServer,
+  gql,
+  UserInputError
+} = require('apollo-server');
+
+const typeDefs = gql`
+  type Query {
+    userWithID(id: ID!): User
+  }
+
+  type User {
+    id: ID!
+    name: String!
+  }
+`;
+
+const resolvers = {
+  Query: {
+    userWithID: (parent, args, context) => {
+      if (args.id <ExpansionPanel 1) {
+        throw new UserInputError('Invalid argument value', {
+          argumentName: 'id'
+        });
+      }
+      // ...fetch correct user...
+    },
+  },
+};
+```
+
+</ExpansionPanel>
+
+This results in a response like the following:
+
+<ExpansionPanel title="Click to expand">
+
+```json{15}
+{
+  "errors": [
+    {
+      "message": "Invalid argument value",
+      "locations": [
+        {
+          "line": 2,
+          "column": 3
+        }
+      ],
+      "path": [
+        "userWithID"
+      ],
+      "extensions": {
+        "argumentName": "id",
+        "code": "BAD_USER_INPUT",
+        "exception": {
+          "argumentName": "id",
+          "stacktrace": [
+            "UserInputError: Invalid argument value",
+            "    at userWithID (/my-project/index.js:25:13)",
+            "    ...more lines...",
+          ]
+        }
+      }
+    }
+  ]
+}
+```
+
+</ExpansionPanel>
+
+## Custom errors
+
+You can create a custom error by defining your own subclass of `ApolloError`, or by initializing an `ApolloError` object directly:
+
+#### Subclass with custom error code
+
+```ts
+import { ApolloError } from 'apollo-server-errors';
+
+export class MyError extends ApolloError {
+  constructor(message: string) {
+    super(message, 'MY_ERROR_CODE');
+
+    Object.defineProperty(this, 'name', { value: 'MyError' });
+  }
+}
+
+throw new MyError('My error message')
+```
+
+#### Direct initialization
+
+```ts
+import { ApolloError } from 'apollo-server-errors';
+
+throw new ApolloError('My error message', 'MY_ERROR_CODE', myCustomDetailsObject);
+```
+
+
+## Masking and logging errors
+
+> To disable stacktraces for production, pass `debug: false` to the Apollo Server constructor or set the `NODE_ENV` environment variable to 'production' or 'test'. Note that this will make the stacktrace unavailable to your application. If you want to log the stacktrace, but not send it in the response to the client, see [Masking and logging errors](#masking-and-logging-errors) below.
```

Comment:
It looks a little weird that this entire subsection is a block quote. Is that intentional?

Also I feel like this could be more from a perspective of "stacktraces are on in dev and off in production, by which we mean NODE_ENV is such and such.  you can override this by specifying debug: true or debug: false explicitly", if that makes sense?

I also have no clue why they are off by default in test, that seems weird and not particularly explained when it was added in https://github.com/apollographql/apollo-server/commit/0e68e2ec4f94dc22d961958c03aea15c6a49120a

Reply:
Ah this must be a block I moved while planning to rework, and then forgot about